cpp/include/cuvs/neighbors/ivf_flat.hpp

```diff
@@ -138,6 +138,10 @@ using list_data = ivf::list<list_spec, SizeT, ValueT, IdxT>;
  */
 template <typename T, typename IdxT>
 struct index : cuvs::neighbors::index {
+  using index_params_type  = ivf_flat::index_params;
```

Comment:
Please also include the other index types (e.g. bfknn, vamana, etc...).

Reply:
I've checked all available indexes and, unfortunately, I cannot add any more instances at this point:
 - bfknn would be the best candidate, but its build/search methods do not adhere to the expected api (missing index/search params as function arguments).
 - vamana and nn_descend do not have corresponding `search` functions, which are needed for instantiating the dynamic batching index.

We can adapt the code to allow bfknn, but it would require more changes. I suggest we postpone this for 25.02 together with https://github.com/rapidsai/cuvs/pull/261#discussion_r1851601263